### ISSUE TITLE
[hub] Invalidate cache when contents of remote repo change

### ISSUE THREAD

**Issue by pcuenca** [huggingface/swift-transformers#39]
We can also do file verification at the same time (something simple: check the file size)

**Comment by FL33TW00D** [comment]
@greenrazer I think we discussed this before right?

**Comment by pcuenca** [comment]
Yes. [This great PR](https://github.com/huggingface/swift-transformers/pull/155) very much goes in this direction and mimics how other libraries work.

I think we can close this issue now. We can also coordinate with @ardaatahan for next steps.

**Comment by FL33TW00D** [comment]
Closing! thanks to @ardaatahan for the great contribution.
